scipy/ndimage/_morphology.py

```diff
@@ -1803,6 +1812,7 @@ def morphological_laplace(input, size=None, footprint=None, structure=None,
         tmp2 = grey_erosion(input, size, footprint, structure, None, mode,
                             cval, origin, axes=axes)
         np.add(tmp1, tmp2, tmp2)
+        input = np.asarray(input)
```

Comment:
You're computing the whole graph leading to `input` three times:
- once in the `np.asarray(input)` call in `grey_dilation`,
- once in the `np.asarray(input)` call in `grey_erosion`,
- and once on this line.

You need to move this line to the top of this function.

Reply:
Thanks, will update.